query/tests/integration.rs

```diff
@@ -222,3 +260,137 @@ EXPLAIN SELECT add_one(1);
     +---------------+--------------------------------------------------------+
     ");
 }
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_strip_indentation_everything_indented() {
+    let query = r#"
+  CREATE FUNCTION add_one()
+  LANGUAGE python
+  AS '
+  def add_one(x: int) -> int:
+    
+    return x + 1
+  ';
+  
+  SELECT add_one(1);
+"#;
+
+    let ctx = SessionContext::new();
+    let component = python_component().await;
+    let formatter = Box::new(StripIndentationFormatter);
+
+    let parser = UdfQueryParser::new(HashMap::from_iter([(
+        "python".to_string(),
+        Lang {
+            component,
+            formatter,
+        },
+    )]));
+    let parsed_query = parser
+        .parse(query, &WasmPermissions::new(), ctx.task_ctx().as_ref())
+        .await
+        .unwrap();
+
+    let df = UdfQueryInvocator::invoke(&ctx, parsed_query).await.unwrap();
+    let batch = df.collect().await.unwrap();
+
+    assert_batches_eq!(
+        [
+            "+-------------------+",
+            "| add_one(Int64(1)) |",
+            "+-------------------+",
+            "| 2                 |",
+            "+-------------------+",
+        ],
+        &batch
+    );
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_strip_indentation_empty_lines_not_indented() {
+    let query = r#"
+  CREATE FUNCTION add_one()
```

Comment:
The fact that the empty line is indented or not is impossible to see in this diff. I suggest that for the indentation tests you don't use multi-line strings but rather an array like this:

```rust
let query = &[
    "CREATE FUNCTION add_one()",
    // or
    "    CREATE FUNCTION add_one()",
    ...
    // this is an non-indended empty line:
    "",
    // and this one would be indended:
    "    ",
];
```